_specs/ecip-1105.md

```diff
@@ -52,177 +43,93 @@ Generally, these constraints give the consumer (the transactor) an ability to ex
 
 ### Definitions
 
-- `chainContext`. The transaction is only valid for blockchain data satisfying ALL OF the annotations.
-- `ANNOTATION_COMPOSITE_PREFIX`. A positive integer between `1` and `0xff` that represents the set of subclass annotations in the `chainContext` (_ie._ _which_ chain context subclasses should the provided values be applied to). This value should be the sum of the subclass' `ANNOTATION_PREFIX`s.
-- `ANNOTATION_PREFIX`s are defined for Subclasses as octal-derived positive integers, limited to the set `2^0,2^1,2^2,2^3,2^4,2^5,2^6,2^7`.
-
-The `chainContext` value should be of the form `ANNOTATION_COMPOSITE_PREFIX || [{subclass value}...]`, where 
-- `...` means "zero or more of the things to the left," and 
-- `||` denotes the byte/byte-array concatenation operator.
-
-The `chainContext` value should be encoded as `ANNOTATION_COMPOSITE_PREFIX || rlp[{subclass value}...]`.
-
 ### Validation
 
-The values defined as subclasses below acts as constraints on transaction validity for specific chain contexts.
+The values defined below act as a constraint on transaction validity.
 Transactions defining constraints which are not satisfied by their chain context should be rejected as invalid.
 Blocks containing invalid transactions should be rejected as invalid themselves, per the _status quo_.
 
-### Subclass Combination
-
-`chainContext` values annotating more than one subclass reference should provide those values in the following sequential order:
-
-1. `ANCESTOR_ID`
-2. `ELIGIBLE_MINER_LIST`
-3. `INELIGIBLE_MINER_LIST`
-4. `EXPIRY`
-
-As above, the `ANNOTATION_COMPOSITE_PREFIX` should be the sum of the designated subclass' `ANNOTATION_PREFIX`s.
-### Subclasses
-
-- An `ANNOTATION_PREFIX` value is used to represent each of the available context subclasses.
-
-#### `ancestorId`
-
-- `ANNOTATION_PREFIX` `1`.
-- `ANCESTOR_ID` `bytes`. A byte array between 4 and 12 bytes in length.
-
-The `ANCESTOR_ID` is a reference to a specific block by concatenating the byte representation of a block number and the first 4 bytes of its hash. 
-The block number's should be encoded as a big endian value and should have left-padding 0's removed.
-The block number value may be omitted in case of reference to the genesis block.
-
-The `ANCESTOR_ID` value should be RLP encoded as a byte array for hashing and transmission.
+#### `ancestorHash`
 
-#### `eligibleMinerList`
+- `ancestorHash` `bytes`. A byte array 32 bytes in length.
 
-- `ANNOTATION_PREFIX` `2`.
-- `ELIGIBLE_MINER_LIST` `[address...]`. A list of addresses.
-- `MAX_ELEMENTS` `3`. The maximum number of addresses that can be provided.
+The `ancestorHash` is a reference to a specific block by hash. 
+If a transaction filling this field is included in block number `n`, the annotated value must match some block numbered `<= n-1` and existing as a direct (canonical; non-ommer) ancestor of the transaction's containing block. 
```

Comment:
Actually wouldn't it be `T ∈ B ⇔ BlockNumber(B) >= BlockNumber(RH)`, since the block containing the subject transaction T will be a greater number than the block cited by the transaction as an `RH`. 

Also this logic-style syntax doesn't include the canonical-ancestry caveat. 

I'm working on a more academic-style paper for this idea that'll have these kinds of terms. For the ECIP, I'm OK leaving it (mostly) in plain English :).